webplugin/js/app/components/answer-feedback-service.js

```diff
@@ -0,0 +1,210 @@
+                    const msgMeta =
+                        this.msgMap[data.msg.key]?.msg?.metadata || {};
+                    msgMeta.KM_ANSWER_FEEDBACK = feedback;
+                } else {
+                    console.error('Failed to submit feedback');
+                }
+            },
+            error: function (data) {
+                console.error(data);
+            },
+        });
+    };
+
+    getPayload = (feedback, data) => {
+        const {
+            msg: { metadata, key },
+            assigneeKey,
+        } = data;
+
+        const answer = this.msgMap[key];
+        let question = answer;
+
+        // find the user message
+        while (
+            !metadata.KM_CONTEXT_QUESTION &&
+            question?.floatWhere !== 'mck-msg-right'
+        ) {
+            question = question.previousSibling;
+        }
+
+        const answerText = answer.msg.message;
+        const questionText = question.msg.message;
+
+        const payload = { feedback, messageKey: key };
+
+        if (!metadata.KM_ANSWER_FEEDBACK) {
+            Object.assign(payload, {
+                source: metadata.KM_ANSWER_SOURCE || 'intent',
+                botKey: assigneeKey,
+                applicationKey: this.options.appId,
+                answer: answerText,
+                question: metadata.KM_CONTEXT_QUESTION || questionText,
+                groupId: CURRENT_GROUP_DATA.tabId,
+                userKey: this.options.userId,
+            });
+        }
+
+        return {
+            payload,
+            method: metadata.KM_ANSWER_FEEDBACK ? 'PATCH' : 'POST',
+        };
+    };
+
+    getActiveClass = (msg, feedback) => {
+        const { KM_ANSWER_FEEDBACK } = msg.metadata;
+
+        return KM_ANSWER_FEEDBACK == feedback ? 'active-feedback' : '';
+    };
+
+    helpFullOnClick = (data) => {
+        kmWidgetEvents.eventTracking(eventMapping.onHelpFulClick);
+        window.Applozic.ALSocket.events.onHelpFulClick({ message: data.msg });
+
+        this.handleOnFeedbackClick(
+            KommunicateConstants.ANSWER_FEEDBACK.HELPFUL,
+            data
+        );
+    };
+
+    notHelpFullOnClick = (data, btn) => {
+        kmWidgetEvents.eventTracking(eventMapping.onNotHelpFulClick);
+        window.Applozic.ALSocket.events.onNotHelpFulClick({
+            message: data.msg,
+        });
+
+        this.handleOnFeedbackClick(
+            KommunicateConstants.ANSWER_FEEDBACK.NOT_HELPFUL,
+            data
+        );
+    };
+
+    handleFeedbackBtnVisible = (msg, floatWhere, group) => {
+        // only visible if the message is from the bot
+        let currentUser = group.users.contact(group.removedUsers || []);
+        currentUser = currentUser[msg.to];
+
+        if (!currentUser) return false;
+
+        if (
+            currentUser.role !==
+            KommunicateConstants.GROUP_ROLE.MODERATOR_OR_BOT
+        ) {
+            return false;
+        }
+
+        return (
+            floatWhere === 'mck-msg-left' &&
```

Comment:
check if it will be impacted by RTL setting, figure out some other way to distinguish it without specifying left or right

Reply:
this will not impact the rtl